airbyte-integrations/integration-test-lib/src/main/java/io/airbyte/integrations/base/TestDestination.java

```diff
@@ -0,0 +1,243 @@
+
+  /**
+   * Function that returns all of the records in destination as json at the time this method is
+   * invoked. These will be used to check that the data actually written is what should actually be
+   * there.
+   *
+   * @param testEnv - information about the test environment.
+   * @return All of the records in the destination at the time this method is invoked.
+   * @throws Exception - can throw any exception, test framework will handle.
+   */
+  protected abstract List<JsonNode> recordRetriever(TestDestinationEnv testEnv) throws Exception;
+
+  /**
+   * Function that performs any setup of external resources required for the test. e.g. instantiate a
+   * postgres database. This function will be called before EACH test.
+   *
+   * @param testEnv - information about the test environment.
+   * @throws Exception - can throw any exception, test framework will handle.
+   */
+  protected abstract void setup(TestDestinationEnv testEnv) throws Exception;
+
+  /**
+   * Function that performs any clean up of external resources required for the test. e.g. delete a
+   * postgres database. This function will be called after EACH test. It MUST remove all data in the
+   * destination so that there is no contamination across tests.
+   *
+   * @param testEnv - information about the test environment.
+   * @throws Exception - can throw any exception, test framework will handle.
+   */
+  protected abstract void tearDown(TestDestinationEnv testEnv) throws Exception;
+
+  @BeforeEach
+  void setUpInternal() throws Exception {
+    Path workspaceRoot = Files.createTempDirectory("test");
+    jobRoot = Path.of(workspaceRoot.toString(), "job");
+    Files.createDirectories(jobRoot);
+    localRoot = Files.createTempDirectory("output");
+    testEnv = new TestDestinationEnv(localRoot);
+
+    setup(testEnv);
+
+    pbf = new DockerProcessBuilderFactory(workspaceRoot, workspaceRoot.toString(), localRoot.toString(), "host");
+  }
+
+  @AfterEach
+  void tearDownInternal() throws Exception {
+    tearDown(testEnv);
+  }
+
+  /**
+   * Verify that when given valid credentials, that check connection returns a success response.
+   * Assume that the {@link TestDestination#getConfig()} is valid.
+   */
+  @Test
+  void testCheckConnection() {
+    // todo (cgardens) - blocked on worker not calling discover worker.
+  }
+
+  /**
+   * Verify that when given invalid credentials, that check connection returns a failed response.
+   * Assume that the {@link TestDestination#getInvalidConfig()} is invalid.
+   */
+  @Test
+  void testCheckConnectionInvalidCredentials() {
+    // todo (cgardens) - blocked on worker not calling discover worker.
+  }
+
+  private static class DataArgumentsProvider implements ArgumentsProvider {
+
+    @Override
+    public Stream<? extends Arguments> provideArguments(ExtensionContext context) {
+      return Stream.of(
+          Arguments.of("exchange_rate_messages.txt", "exchange_rate_catalog.json")
```

Comment:
why is there something about exchange_rate_message & stripe in this class?

Reply:
there are names of sample datasets we run the tests against. i've used the names of what generated them so i can keep track of them, but we can replace them with any descriptive name.